src/esssans/i_of_q.py

```diff
@@ -178,10 +190,21 @@ def merge_spectra(
             data_dims.remove(dim)
             data_dims.append(dim)
         stripped = stripped.transpose(data_dims)
-        # Flatten to Q such that `hist` below will turn this into the new Q dim.
-        flat = stripped.flatten(dims=to_flatten, to='Q')
+        # Flatten to helper dim such that `hist` will turn this into the new Q dim(s).
+        # For sc.hist this has to be named 'Q'.
+        helper_dim = 'Q'
+        flat = stripped.flatten(dims=to_flatten, to=helper_dim)
 
-        out = flat.hist(**edges)
+        if len(edges) == 1:
+            out = flat.hist(**edges)
+        else:
+            # sc.hist (or the underlying sc.bin) cannot deal with extra data dims,
+            # work around by flattening and regrouping.
+            out = (
+                flat.flatten(to=str(uuid.uuid4()))
+                .group(*[flat.coords[dim] for dim in flat.dims if dim != helper_dim])
+                .hist(**edges)
+            )
```

Comment:
For my understanding, can you explain how this works for Qxy. `helper_dim` is always `Q`, so I'm not really following what is happening.

Reply:
The dims that were "kept" have been flattened and regrouped, but we do not want to group in the dim that we will `hist` afterwards.

Reply:
I am wondering if we should just convert the denominator to "events". This would unify a bunch of handling in the merge and normalization. I will consider this separately, not in this PR.

Reply:
> I am wondering if we should just convert the denominator to "events"

How do you do that? Make one event per bin and give it the counts in the bin as weights? (And by bin I mean the original binning of the monitor data)

Reply:
Yes, exactly.